client/src/components/Header.jsx

```diff
@@ -1,37 +1,84 @@
 import React from 'react';
-/*
-still need to solve:
-- logic for displaying features with value 'true'
-- why can't I get nested properties? (props.headerData.location.city)
-- urls for icons
-*/
-const Header = (props) => (
-  <div className="listing-header">
-    <div className="headline">
-      <h1>{props.headline}</h1>
-    </div>
-    <div className="header-widgets">
-      <div className="header-info">
-        <div className="stars-info">
-          {props.stars} ({props.reviews}) •
-        </div>
-        <div className="superhost-badge">
-          {props.superHost === true ? 'Superhost •' : null}
-        </div>
-        <div className="location-info">
-          {props.location ? props.location : null}
-        </div>
-      </div>
-      <div className="header-actions">
-        <div className="share-button">
-          <img className="share-icon" src="http://localhost:3005/assets/share.svg" />Share
+import axios from 'axios';
+
+const API_URL = 'http://localhost:3005';
+
+class Header extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      listingId: this.props.id,
+      headline: 'loading',
+      stars: 0,
+      reviews: 0,
+      superHost: false,
+      location: 'loading'
+    };
+  }
+
+  fetchHeaderData(id) {
```

Comment:
I haven't gotten this far in my project, but I wonder if you can set it up so that one parent component does the fetch request and passes down the data, instead of having multiple components fetch the data.